db/migrate/20180806151925_add_subnet_name_unique_constraint.rb

```diff
@@ -0,0 +1,58 @@
+class AddSubnetNameUniqueConstraint < ActiveRecord::Migration[5.1]
+  def up
+    deduplicate_subnets
+    # we need to specify the lenght to make MySQL happy
+    add_index :subnets, :name, :unique => true, length: { name: 255 }
+  end
+
+  def down
+    remove_index :subnets, :name
+  end
+
+  private
+
+  def deduplicate_subnets
+    duplicate_names = Subnet.unscoped.select(:name).group(:name).having('count(name) > 1').pluck(:name)
+    Rails.logger.info("Found #{duplicate_names} duplicate subnet names.")
```

Comment:
Use `say` or this could be candidate for `say_with_time`: https://rails-bestpractices.com/posts/2010/08/19/use-say-and-say_with_time-in-migrations-to-make-a-useful-migration-log/

Reply:
fixed